rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -160,7 +168,7 @@ fn test_check_address() {
             address: "n47QBape2PcisN2mkHR2YnhqoBr56iPhJh".to_string(),
         },),
     );
-    assert!(result.is_err_and(|err| format!("{:?}", err).contains("Not a bitcoin mainnet address")));
+    assert!(result.is_err_and(|err| format!("{:?}", err).contains("Not a bitcoin address")));
```

Comment:
This is not correct as it is simply not a mainnet address. The error message should depend on the expected network type.

Could you also add a test that a mainnet address is not accepted when using the `testnet` network?

Reply:
> Could you also add a test that a mainnet address is not accepted when using the testnet network?

+1 for this

> This is not correct as it is simply not a mainnet address.

I'm confused @THLO by this, because this part of the test is exactly for testing a testnet address, no?
```rust
// Test with an testnet address
    let result = query_candid::<_, (CheckAddressResponse,)>(
        &env,
        kyt_canister,
        "check_address",
        (CheckAddressArgs {
            address: "n47QBape2PcisN2mkHR2YnhqoBr56iPhJh".to_string(),
        },),
    );
    assert!(result.is_err_and(|err| format!("{:?}", err).contains("Not a bitcoin address")));
```

Reply:
> Could you also add a test that a mainnet address is not accepted when using the testnet network?

Good suggestion! Just added such a test.

> This is not correct as it is simply not a mainnet address. 

Because the `bitcoin::Network::Bitcoin` variant would print as "bitcion", so it is testing a testnet address against mainnet. 

The newly added test of checking mainnet address against testnet setup would match the string `contains("Not a testnet address")`.

Reply:
After implementing our own type `BtcNetwork` to restrict to only 2 variants `mainnet` and `testnet`, it should be clearer with matching strings like "not a bitcoin testnet address".